account_invoice_validation/models/account_move.py

```diff
@@ -0,0 +1,450 @@
+        :param validation_state: str
+        :return: bool
+        """
+        types = self.get_concerned_types()
+        invoices = self.filtered(lambda i: i.move_type not in types)
+        if invoices:
+            details = "\n- ".join(invoices.mapped("display_name"))
+            message = (
+                _("These documents are not supplier " "invoice/refund:\n- %s") % details
+            )
+            raise ValidationError(message)
+        return True
+
+    def _update_validation_state(self, values: dict, validation_state: str) -> bool:
+        """
+        Update the validation state with the given value
+        We can put some check before the write if necessary
+        :param validation_state: str
+        :return: bool
+        """
+        self._check_before_update_validation_state(validation_state)
+        return self.write(values)
+
+    def action_block_state(self):
+        """
+        action calling the wizard account_move_note from block button
+        """
+        formid = self.env.ref(
+            "account_invoice_validation.account_move_note_form_view"
+        ).id
+        context = self.env.context.copy()
+        context.update({"movingToState": "blocked"})
+        return {
+            "type": "ir.actions.act_window",
+            "view_type": "form",
+            "view_mode": "form",
+            "res_model": "account.move.note",
+            "view_id": formid,
+            "target": "new",
+            "context": context,
+        }
+
+    def action_block_state_continue(self, note: str, pending_date: str) -> dict:
+        """
+        Update the validation state to locked
+        :param pending_date: str (date)
+        :param note: str
+        :return: dict
+        """
+        self._check_pending_date(pending_date)
+        if not note:
+            message = _("Please specify a note!")
+            raise UserError(message)
+        validation_state = "locked"
+        values = {
+            "validation_state": validation_state,
+            "pending_date": pending_date,
+        }
+        self._update_validation_state(values, validation_state)
+        for rec in self:
+            rec.message_post(body=note)
+        return {}
+
+    def _check_pending_date(self, pending_date: str) -> bool:
+        """
+        If the validity of the given pending date
+        :param pending_date: str (date)
+        :return: bool
+        """
+        if not pending_date:
+            message = _("Please set a pending date!")
+            raise UserError(message)
+        real_date = fields.Date.from_string(pending_date)
+        if real_date <= fields.Date.from_string(fields.Date.today()):
+            message = _("Please set a pending date greater than today!")
+            raise UserError(message)
+        return True
+
+    def action_assign(self) -> dict:
+        """
+        action calling the wizard account_move_note from assign button
+        """
+        formid = self.env.ref(
```

Comment:
These lines seems the same each time.
Maybe you should refactor them into a single function and then edit what you need (the `context`).